src/modules/endpoint/DataContainer.js

```diff
@@ -0,0 +1,238 @@
+   * existent. Initializes non persisted fields with null.
+   * @param cityCode
+   * @param languageCode
+   */
+  setContext: (cityCode: string, languageCode: string) => Promise<void>,
+
+  /**
+   * Returns an Array of CityModels.
+   * @throws Will throw an error if the array is null.
+   */
+  getCities: () => Promise<Array<CityModel>>,
+
+  /**
+   * Sets the cities but does not persist them.
+   * TODO: Offline available cities will be persisted in NATIVE-175. For now switching cities when offline is not possible.
+   * @param cities
+   */
+  setCities: (cities: Array<CityModel>) => Promise<void>,
+
+  /**
+   * Returns an Array of LanguageModels.
+   * @throws Will throw an error if the array is null.
+   */
+  getLanguages: () => Promise<Array<LanguageModel>>,
+
+  /**
+   * Sets the languages and persists them.
+   * @param languages
+   */
+  setLanguages: (languages: Array<LanguageModel>) => Promise<void>,
+
+  /**
+   * Returns the CategoriesMapModel.
+   * @throws Will throw an error if the CategoriesMapModel is null.
+   */
+  getCategoriesMap: () => Promise<CategoriesMapModel>,
+
+  /**
+   * Sets the categories and persists them.
+   * @param categories
+   */
+  setCategoriesMap: (categories: CategoriesMapModel) => Promise<void>,
+
+  /**
+   * Returns an Array of events.
+   * @throws Will throw an error if the array is null.
+   */
+  getEvents: () => Promise<Array<EventModel>>,
+
+  /**
+   * Sets the events and persists them.
+   * @param events
+   */
+  setEvents: (events: Array<EventModel>) => Promise<void>,
+
+  /**
+   * Returns the ResourceCache.
+   * @throws Will throw an error if the ResourceCache is null.
+   */
+  getResourceCache: () => Promise<ResourceCacheStateType>,
+
+  /**
+   * Sets the ResourceCache and persists it.
+   * @param resourceCache
+   */
+  setResourceCache: (resourceCache: ResourceCacheStateType) => Promise<void>,
+
+  /**
+   * Returns whether the CategoriesMap has been loaded or not.
+   */
+  categoriesAvailable (): boolean,
+
+  /**
+   * Returns whether the languages have been loaded or not.
+   */
+  languagesAvailable (): boolean,
+
+  /**
+   * Returns whether the ResourceCache have been loaded or not.
+   */
+  resourceCacheAvailable (): boolean,
+
+  /**
+   * Returns whether the events have been loaded or not.
+   */
+  eventsAvailable (): boolean
+}
+
+class DataContainer implements DataContainerInterface {
+  _databaseConnector: DatabaseConnector
+  _context: DatabaseContext | null = null
+
+  _cities: Array<CityModel> | null
+  _categoriesMap: CategoriesMapModel | null
+  _languages: Array<LanguageModel> | null
+  _resourceCache: ResourceCacheStateType | null
+  _events: Array<EventModel> | null
+
+  constructor () {
+    this._databaseConnector = new DatabaseConnector()
+  }
+
+  getCities = async (): Promise<Array<CityModel>> => {
+    if (this._cities === null) {
+      throw Error('Cities are null.')
```

Comment:
I think we can continue to live with the exceptions but this is not a perfect design. By perfect I mean that it is not possible to miss-use the class.

Reply:
I was unable to think of a better design though :/